app/src/main/java/me/saket/dank/urlparser/UrlParser.java

```diff
@@ -261,13 +264,23 @@ public ImgurLink createImgurLink(String url, @Nullable String title, @Nullable S
         .scheme("https")
         .build();
 
-    // Attempt to get direct links to images from Imgur submissions.
-    // For example, convert 'http://imgur.com/djP1IZC' to 'http://i.imgur.com/djP1IZC.jpg'.
-    if (!isImageOrGifUrlPath(url) && !url.endsWith("mp4")) {
-      // If this happened to be a GIF submission, the user sadly will be forced to see it instead of its GIFV.
-      imageUrl = imageUrl.newBuilder(imageUrl.encodedPath() + ".jpg")
-          .host("i.imgur.com")
-          .build();
+    String imageUrlPath = imageUrl.encodedPath();
+
+    if (!url.endsWith("mp4")) {
+      if (isImageOrGifUrlPath(imageUrlPath)) {
+        // Strip any preview-related suffixes and queries
+        imageUrl = imageUrl.newBuilder()
+            .encodedQuery(null)
+            .encodedPath(imgurPreviewPat.matcher(imageUrlPath).replaceFirst("."))
```

Comment:
It probably won’t occur but what if that pattern was found earlier than before extension? Maybe it’s better to replace last occurrence instead.

Reply:
Using java's matcher this can only be done by a more precise regex. Something like `s/_d(\.\w+)/$1/`